lib/services/auth/auth_service.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:convert';
+
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:survey/gen/configs.gen.dart';
+import 'package:survey/services/locator/locator_service.dart';
+import 'package:survey/models/auth_token_info.dart';
+import 'package:object_mapper/object_mapper.dart';
+import 'package:survey/models/user_info.dart';
+import 'package:survey/services/api/api_service.dart';
+import 'package:survey/services/http/http_service.dart';
+import 'package:survey/services/user/user_service.dart';
+
+part 'params/auth_login_params.dart';
+
+abstract class AuthService {
+  static const loginEndPoint = "/oauth/token";
+  static const logoutEndPoint = "/oauth/revoke";
+  static const preferenceTokenKey = "auth_service_preference_token";
+
+  bool get isAuthenticated;
+
+  String? get token;
+
+  UserInfo? get user;
```

Comment:
let's not mix all of this:
- prefernce is your local storage preference, 
- whereas we are managing these AuthService as to communicate to the remote backend, like you're having in the other Service.
- isAuthenticated or not is usually managed separately from the remote config as this is a common state managed by the app itself.
- Token is also manageable by another place where you could retrieve/set differently. If you want to manipulate the logic on the top of it i.e to define login/non-login status, it shouldn't be coupled with the remote API invoker logics i.e calling to login/logout...

From my experience, I'd say this implementation violates quite seriously the separation of concern principle.